### ISSUE TITLE
[Event Hubs Client] Test Suite Improvements

### ISSUE THREAD

**Issue by jsquire** [Azure/azure-sdk-for-net#5982]
# Summary

Some of the Event Hubs client library Live tests have a measure of non-determinism in them due to timing issues, differences in test runner behavior between the full .NET framework and the .NET Core runner, and what are believed to be some race conditions in asynchronous tests around shared diagnostics and activity tracing.  In addition, the tests currently share a common namespace and event hub which prevents running them in parallel and potentially allows misbehaving or failing tests to influence others.

The intention of this epic is to capture the identified opportunities to improve the Event Hubs client library tests, adding isolation, resilience, and reliability to the suite and enabling speedier runs with support for parallel execution.

# Scope of Work

- Improve the existing test suite across the different target platforms and test runners, with a goal of ensuring that runs are consistent and reliable.

- Increase the isolation between tests, attempting to ensure that each test is entirely self-contained and not subject to environmental considerations or state shared with other tests, including any Azure service artifacts used.

- Empower tests to create and remove Azure service artifacts that they rely on, eliminating the need to have a fully prepared environment prior to runs and control costs.  Embrace the minimal required state needed to allow for tests to manage themselves.

- Tests should produce deterministic results, with no known issues of intermittent failures due to timing or other causes.  It is reasonable to have tests that suffer from the occasional intermittent failure, though these should be the exception and not the rule.

# Out of Scope

- Refactoring of the general approach for testing, the test cases, code style, and other artifacts of the code base unrelated to timing-related items.

- Adding of additional test coverage or new testing scenarios to cover additional features of the Event Hubs client library.

# Success Criteria

- Retain a focus on ensuring that the ramp-up time and requirements for contributors to execute the test suite remain minimal and approachable; there should be no special knowledge or efforts require to execute tests after cloning that isn't approachable by those with minimal Azure experience.

- All steps to executing tests should be well-documented in the repository `ReadMe` and be approachable to those running the suite for the first time.

- Tests should be able to run in parallel safely or should be organized into specific units of tests that must be run serialized.  Ideally, the test suite should be able to fully run within a 10 minute window consistently.

- Tests should be executable on each target platform and operating system successfully, while producing consistent and repeatable results on each.

- Tests should be reliable, producing consistent results across runs without intermittent failures under normal conditions.

- Tests should be isolated from one another; test failures or misbehaving tests should not be able to exert influence or cascade failures across other tests.

- Tests should clean up after themselves, leaving the environment in the same state that it was in before the test run.   Any unrecoverable failures should be identifiable and have an associated clean-up utility that can be run periodically to keep the environment in a good state.
